### DISCUSSION TITLE
Support both regexp and invert regexp in same invocation

### DISCUSSION THREAD

**Discussion by balki** [BurntSushi/ripgrep#2175] [General] [answered]
Suppose a text file contains below:

    ❯ cat foo.txt 
    hello world
    hello foo
    foo bar

consider this:

    ❯ rg hello
    foo.txt
    1:hello world
    2:hello foo

Say, I don't want lines that match world, I have to do below:

    ❯ rg hello | rg -v world
    foo.txt:hello foo

This requires two `rg` invocation and loose all the colors. It would be nice to support both in single invocation.

    ❯ rg -e hello -y world
    error: Found argument '-y' which wasn't expected, or isn't valid in this context

**Comment by BurntSushi** [comment, marked as answer]
This is a duplicate of #875. 

`rg foo | rg -v bar` is otherwise the intended solution. In general, "losing colors and formatting" is not something I can allow to justify eschewing the utility of shell pipelines in all cases. So I don't know if #875 will ever happen.

Running ripgrep multiple times should in general be a non-issue. At least, I do not consider multiple invocations in and of itself a downside.

**Reply by balki** [reply]
Thank you for your response. I understand the use-case is not justified for the feature. Just in case you revisit this later, here are the list of downsides I can think of.

1. No colors and formatting
2. The filename becomes part of the text and affects the subsequent searches. `--no-filename` flag can be used. But now the filenames are lost.
3. line numbers are lost

**Reply by BurntSushi** [reply]
You can get colors back with `--color always` and line number back with `-n`. So `rg foo -n --color always | rg -v bar`.

(2) is indeed a downside, but one that is usually pretty easy to work around.
